### PR TITLE
Lp 86 fix replay button

### LINKED ISSUE #86: Fix Replay button
**Is your feature request related to a problem? Please describe.**
When you hit the replay button on the modal menu nothing happens

**Describe the solution you'd like**

when you hit the replay button the game get reset and the entities get  set to their initial positions

### PR DESCRIPTION
**Pull request by Lap343**
## Changes
1. Have the replay button to do two different things depending on what game mode you are on.
2. If you are doing the main game mode the replay button re-shuffles the games and restarts the index
3. If you are from the practice game screen you will replay the game you are on

## Purpose
So the player can replay the game

Closes #86

### CODE REVIEW

**Review by connielion — APPROVED** [review]
Noice!

**Review by JorgeLVilla — APPROVED** [review]
Nice job Luis!

**Review by Gjhnsn — APPROVED** [review]
Looks good, nice feature to add

File: components/MenuModal.js [hunk 1]
```diff
@@ -47,7 +54,17 @@ const MenuModal = ({ navigation }) => {
...
+                                navigation.reset({
+                                    index: 0,
+                                    routes: [
+                                        { name: 'Main Menu' },
+                                        {
+                                            name: 'Random Game Select',
+                                            params: { fromGameSelect: false },
+                                        },
+                                    ],
+                                })
```
Comment (kmausolf) [inline components/MenuModal.js:67]: Neat!  This seems like a pretty concise way to know where you came from.

Comment (Lap343) [reply components/MenuModal.js:67]: When resetting the stack and for the .goBack() func to work you have to specify all the routes

**Pull request merged**